slyd/slyd/utils/storage.py

```diff
@@ -0,0 +1,199 @@
+import os
+import shutil
+
+from six.moves import StringIO
+from six.moves.urllib.parse import urljoin
+
+from django.core.files.move import file_move_safe
+from django.core.files.storage import Storage, FileSystemStorage, File
+from dulwich.objects import Blob
+
+
+FILE_MODE = 0o100644
+CHANGE_UPDATE = 0
+CHANGE_RENAME = 1
+CHANGE_DELETE = 2
+
+
+class CommitingStorage(object):
+    def get_available_name(self, name, max_length=None):
+        return name
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self):
+        pass
+
+    def checkout(self):
+        pass
+
+    def commit(self):
+        pass
+
+
+class FsStorage(CommitingStorage, FileSystemStorage):
+    def isdir(self, name):
+        return os.path.isdir(self.path(name))
+
+    def isfile(self, name):
+        return os.path.isfile(self.path(name))
+
+    def move(self, old_file_name, new_file_name, allow_overwrite=False):
+        file_move_safe(self.path(old_file_name), self.path(new_file_name),
+                       allow_overwrite=True)
+
+    def makedirs(self, name):
```

Comment:
do we need this method? FileSystemStorage automatically makes directories in _save, and GitStorage doesn't need it